ocs_ci/ocs/ui/base_ui.py

```diff
@@ -1485,20 +1500,1136 @@ def __init__(self):
+                "instead use nav_into_node(node_name_option='node_name')"
+            )
+
+        if node_index_option is not None:
+            loc = (
+                self.topology_df.at[node_index_option, "navigate_into_xpath"],
+                By.XPATH,
+            )
+            logger.info(f"Open node by index {node_index_option}")
+        elif node_name_option is not None:
+
+            filtered_line = self.topology_df[
+                self.topology_df["entity_name"] == node_name_option
+            ]
+            loc = (filtered_line["navigate_into_xpath"].iloc[0], By.XPATH)
+            logger.info(f"Open node by name {node_name_option}")
+        else:
+            raise IncorrectUiOptionRequested(
+                f"Pass one of required options to use method '{self.nav_into_node.__name__}'"
+            )
+        self.do_click(loc, 60, True)
+        self.page_has_loaded(5, 5, self.topology_loc["topology_graph"])
+        return OdfTopologyDeploymentsView()
+
+    def record_nested_deployments(
+        self, node_index: int, df: pd.DataFrame
+    ) -> pd.DataFrame:
+        """
+        Records nested deployments for a specific node in the Topology.
+
+        Args:
+            node_index (int): Index of the node in the Topology.
+            df (pd.DataFrame): DataFrame representing the nested deployments.
+
+        Returns:
+            pd.DataFrame: Updated Topology DataFrame.
+
+        Example:
+            record_nested_deployments(0, nested_deployments_df)
+            # Returns the updated Topology DataFrame with nested deployments recorded.
+
+        """
+        self.topology_df.at[node_index, "nested_deployments"] = df
+        return self.topology_df
+
+    def read_details(self) -> dict:
+        """
+        Reads and retrieves details of a node from the UI.
+
+        Returns:
+            dict: Dictionary containing the node details.
+
+        Raises:
+            IncorrectUiOptionRequested: If the wrong level of topology is opened instead of the Node level.
+
+        Example:
+            read_details()
+            # Returns {'name': 'node-1', 'status': 'Ready', 'role': 'worker', 'operating_system': 'linux',
+            #          'kernel_version': '4.18.0-305.12.1.el8_4.x86_64', 'instance_type': 'm5.large',
+            #          'OS_image': 'CentOS Linux 8 (Core)', 'architecture': 'amd64',
+            #          'addresses': 'External IP: 203.0.113.10; Hostname: node-1; Internal IP: 192.168.0.1',
+            #          'kubelet_version': 'v1.21.2', 'provider_ID': 'aws', 'annotations_number': '5 annotations',
+            #          'external_id': '-', 'created': 'Jun 1, 2023, 10:00 AM'}
+
+        """
+        details_dict = dict()
+        if (
+            self.get_element_text(self.topology_loc["details_sidebar_entity_header"])
+            == "Node details"
+        ):
+            filtered_dict = {
+                k: v
+                for k, v in self.topology_loc.items()
```

Comment:
Can you use it with relevant var names? and not key value

Reply:
done